src/Mod/Draft/draftguitools/gui_selectplane.py

```diff
@@ -186,7 +186,14 @@ def handle(self):
         sel = FreeCADGui.Selection.getSelectionEx()
         if len(sel) == 1:
             sel = sel[0]
-            if Draft.getType(sel.Object) == "Axis":
+            if hasattr(sel.Object, 'TypeId') and sel.Object.TypeId == 'App::Part':
+                FreeCAD.DraftWorkingPlane.setFromPlacement(sel.Object.getGlobalPlacement(), rebase=True)
```

Comment:
So the idea is just to move the grid to the position of Part?

Reply:
yes, nothing more than this :)